modules/library/main/src/main/java/org/geotools/data/util/ArrayConverterFactory.java

```diff
@@ -55,4 +58,18 @@ public Converter createConverter(Class<?> source, Class<?> target, Hints hints)
             return (T) result;
         }
     }
+
+    private static class ArrayToArrayConverter implements Converter {
+        @Override
+        public <T> T convert(Object source, Class<T> target) throws Exception {
+            int arrLength = Array.getLength(source);
+            Class<?> elementType = target.getComponentType();
+            Object result = Array.newInstance(elementType, arrLength);
+
+            for (int count = 0; count < arrLength; count++) {
+                Array.set(result, count, Converters.convert(Array.get(source, count), elementType));
```

Comment:
This looks great, is it worth checking for `null` here (returned when converters cannot make the conversion) or do you want to just let it ride?

Reply:
@jodygarnett good catch. I think we could check the source element, if it's null then set the target array element to null, if the source is not null and the converted is (error in conversion) we can return null for the whole target array (to notify a conversion error). What do you think?

Reply:
Yes that is the correct approach.

Note: We did experiment with returning exceptions for this kind of failure, but the resulting exceptions would end up user visible (resulting in confused unhappy complaints). We ended up with this *return null* approach which IMHO ends up being too silent on details ... a tradeoff.